### PR TITLE
#65: Zone detail screen

### PR DESCRIPTION
**Pull request by lukybil**
- Zone detail screen
- New way of passing zone and point data between screens using global store
- type names cleanup

### CODE REVIEW

File: app/zone-details/index.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+const ZoneDetailsScreen = () => {
+  // reason for not passing the whole object: https://reactnavigation.org/docs/params/#what-should-be-in-params
+  const zoneDetailsParams = useLocalSearchParams<ZoneDetailsParamas>()
+  const t = useTranslation()
+
+  const id = zoneDetailsParams.id ? Number.parseInt(zoneDetailsParams.id, 10) : -1
+
+  const zone = useMapZone(id, true)
+
+  if (!zone) {
```
Comment (lukybil) [inline app/zone-details/index.tsx:26]: I know that it is a negative condition, but it saves indentation and makes it clearer what happens, for return value of Elements, I would do it this way

File: app/zone-details/index.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+const ZoneDetailsScreen = () => {
+  const zoneDetailsParams = useLocalSearchParams<ZoneDetailsParamas>()
+  const globalStore = useContext(GlobalStoreContext)
+
+  const id = Number.parseInt(zoneDetailsParams.id ?? '0', 10)
+
+  const zone = useMemo(
+    () => globalStore.mapFeatures?.get(id) as Feature<Polygon, SelectedUdrZone>,
```
Comment (radoslavzeman) [inline app/zone-details/index.tsx]: it would be safer without casting, if possible

File: state/GlobalStoreProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+type ContextProps = {
+  mapFeatures: MapFeatureHashMap | null
+  setMapFeatures: Dispatch<SetStateAction<MapFeatureHashMap | null>>
+}
+
+export const GlobalStoreContext = createContext({} as ContextProps)
+
+type Props = {
+  children: React.ReactNode
```
Comment (radoslavzeman) [inline state/GlobalStoreProvider.tsx:18]: We write just `ReactNode` and import it.

File: state/GlobalStoreProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+const GlobalStoreProvider = ({ children }: Props) => {
```
Comment (radoslavzeman) [inline state/GlobalStoreProvider.tsx:21]: But you can also use just `PropsWithChildren` from react.

File: app/zone-details/index.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import { SelectedUdrZone } from '@/modules/map/types'
+import { GlobalStoreContext } from '@/state/GlobalStoreProvider'
+
+type ZoneDetailsParamas = {
+  id: string
+}
+
+const ZoneDetailsScreen = () => {
+  const zoneDetailsParams = useLocalSearchParams<ZoneDetailsParamas>()
+  const globalStore = useContext(GlobalStoreContext)
```
Comment (radoslavzeman) [inline app/zone-details/index.tsx]: Please create `useGeneralStoreContext` for this :)

**Review by radoslavzeman — APPROVED** [review]
Nice :)

File: modules/map/hooks/useNormalizedPoint.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import { NormalizedPoint, SelectedPoint } from '@/modules/map/types'
+import { normalizePoint } from '@/modules/map/utils/normalizePoint'
+
+export function useNormalizedPoint<Point extends SelectedPoint | undefined | null>(
+  point: Point,
+): Point extends undefined | null ? NormalizedPoint | null : NormalizedPoint
+export function useNormalizedPoint(
+  point: SelectedPoint | undefined | null,
+): NormalizedPoint | null {
+  const [, i18n] = useTranslation()
```
Comment (radoslavzeman) [inline modules/map/hooks/useNormalizedPoint.ts:13]: You can also use `useLocale()` custom hook we have: `const locale = useLocale()`. This is also how other intl libraries work. 
Or if you need i18n instance, we usually do `const { i18n } = useTranslation()`

File: state/hooks/useMapZone.ts [hunk 7]
```diff
@@ -1,29 +1,36 @@
...
+  | (NormalizePropertiesOnly extends true ? NormalizedUdrZone : Feature<Polygon, SelectedUdrZone>)
+  | null
 export function useMapZone(
   id: number,
-  propertiesOnly: boolean = false,
-): SelectedUdrZone | Feature<Polygon, SelectedUdrZone> | null {
+  normalizedPropertiesOnly: boolean = false,
+): NormalizedUdrZone | Feature<Polygon, SelectedUdrZone> | null {
   const globalStore = useContext(GlobalStoreContext)
+  const [, i18n] = useTranslation()
```
Comment (radoslavzeman) [inline state/hooks/useMapZone.ts:18]: useLocale

File: app/zone-details/index.tsx [hunk 8]
```diff
@@ -19,7 +19,7 @@ const ZoneDetailsScreen = () => {
   const zoneDetailsParams = useLocalSearchParams<ZoneDetailsParamas>()
   const t = useTranslation()
 
-  const id = Number.parseInt(zoneDetailsParams.id ?? '0', 10)
+  const id = zoneDetailsParams.id ? Number.parseInt(zoneDetailsParams.id, 10) : -1
```
Comment (radoslavzeman) [inline app/zone-details/index.tsx:22]: null instead of -1?

**Review by radoslavzeman — APPROVED** [review]

**Pull request merged**